src/main/kotlin/io/perfometer/data/csv/CsvParser.kt

```diff
@@ -0,0 +1,61 @@
+package io.perfometer.data.csv
+
+internal class CsvParser(private val csvText: String, private val delimiter: Char = ',') {
+
+    private var position = 0
+
+    fun parse(): List<List<String>> {
+        position = 0
+        val result = mutableListOf<List<String>>()
+        while (notFinished()) {
+            result.add(parseRecord())
+        }
+        return result
+    }
+
+    private fun parseRecord(): List<String> {
+        val result = mutableListOf<String>()
+        while (checkCurrentCharIsNot('\n')) {
+            result.add(parseField())
+        }
+        while (checkCurrentCharIs('\n', '\r')) position++
```

Comment:
See my previous comment, I'd use the same method `isChar(...)` in both cases, for positive and negative check.